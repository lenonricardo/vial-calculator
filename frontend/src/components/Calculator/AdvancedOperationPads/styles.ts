import styled from 'styled-components'
import { Pad } from '../NumberPads/styles'

export const TopPadWrapper = styled.div`
  display: flex;
  flex-direction: row;
  transform: translateY(var(--pad-width));
  margin-top: calc(-1 * var(--pad-width));
  margin-right: var(--pad-width);
`

export const TopPad = styled(Pad)`
  background-color: var(--secondary-operations);
`
