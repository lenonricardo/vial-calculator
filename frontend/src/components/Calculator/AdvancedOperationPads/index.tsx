import { TopPadWrapper, TopPad } from './styles'

interface OperationPadsProps {
  onInput: (input: string) => void
}

enum AdvancedOperationEnum {
  ALL_CLEAR = 'AC',
  SQUARE_ROOT = '√',
  EXPONENTIATION = 'xʸ',
  PERCENTAGE = '%'
}

export default function AdvancedOperationPads({ onInput }: OperationPadsProps) {
  const advancedOperationPads: AdvancedOperationEnum[] = Object.values(
    AdvancedOperationEnum
  )

  const handleOperationClick = (operation: string) => {
    onInput(operation)
  }

  return (
    <TopPadWrapper>
      {advancedOperationPads.map((label) => (
        <TopPad key={label} onClick={() => handleOperationClick(label)}>
          {label}
        </TopPad>
      ))}
    </TopPadWrapper>
  )
}
