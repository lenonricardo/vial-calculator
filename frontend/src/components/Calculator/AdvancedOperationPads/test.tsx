import { render } from '@testing-library/react'

import AdvancedOperationPads from '.'

describe('<AdvancedOperationPads />', () => {
  it('should render the component', () => {
    const mockOnInput = jest.fn()
    const mockOnClear = jest.fn()

    const { container } = render(
      <AdvancedOperationPads onInput={mockOnInput} onClear={mockOnClear} />
    )

    expect(container.firstChild).toMatchSnapshot()
  })
})
