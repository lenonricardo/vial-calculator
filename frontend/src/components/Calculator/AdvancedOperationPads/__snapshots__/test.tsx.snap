// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AdvancedOperationPads /> should render the component 1`] = `
.c1.c1 {
  width: var(--pad-width);
  height: var(--pad-height);
  color: var(--font-pads);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: var(--default-font-size);
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  border-radius: 0;
}

.c1.c1:hover {
  background-color: #fff;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-transform: translateY(var(--pad-width));
  -ms-transform: translateY(var(--pad-width));
  transform: translateY(var(--pad-width));
  margin-top: calc(-1 * var(--pad-width));
  margin-right: var(--pad-width);
}

.c2.c2 {
  background-color: var(--secondary-operations);
}

.c2.c2:hover {
  background-color: var(--secondary-operations);
}

<div
  class="c0"
>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium c1 c2 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    AC
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium c1 c2 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    √
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium c1 c2 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    xʸ
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium c1 c2 css-1e6y48t-MuiButtonBase-root-MuiButton-root"
    tabindex="0"
    type="button"
  >
    %
    <span
      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
    />
  </button>
</div>
`;
